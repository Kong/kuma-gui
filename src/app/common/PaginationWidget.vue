<template>
  <div class="pagination">
    <KButton
      v-if="props.hasPrevious"
      appearance="primary"
      data-testid="pagination-previous-button"
      @click="onPreviousButtonClick"
    >
      <KIcon
        icon="chevronLeft"
        color="currentColor"
        size="16"
        hide-title
      />

      Previous
    </KButton>

    <KButton
      v-if="props.hasNext"
      appearance="primary"
      data-testid="pagination-next-button"
      @click="onNextButtonClick"
    >
      Next

      <KIcon
        icon="chevronRight"
        color="currentColor"
        size="16"
        hide-title
      />
    </KButton>
  </div>
</template>

<script lang="ts" setup>
import { datadogLogs } from '@datadog/browser-logs'
import { KButton, KIcon } from '@kong/kongponents'

import { datadogLogEvents } from '@/utilities/datadogLogEvents'

const props = defineProps({
  hasPrevious: {
    type: Boolean,
    default: false,
  },

  hasNext: {
    type: Boolean,
    default: false,
  },
})

const emit = defineEmits(['next', 'previous'])

function onNextButtonClick() {
  emit('next')
  datadogLogs.logger.info(datadogLogEvents.PAGINATION_NEXT_BUTTON_CLICKED)
}

function onPreviousButtonClick() {
  emit('previous')
  datadogLogs.logger.info(datadogLogEvents.PAGINATION_PREVIOUS_BUTTON_CLICKED)
}
</script>

<style lang="scss" scoped>
.pagination {
  padding: var(--spacing-sm);
  text-align: right;

  button {
    font-size: 14px;
    padding: 4px 8px;

    &:not(:first-of-type) {
      margin-left: var(--spacing-sm);
    }

    &:after {
      display: none;
    }
  }
}
</style>
