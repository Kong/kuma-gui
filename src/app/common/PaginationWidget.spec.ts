import { mount } from '@vue/test-utils'
import { datadogLogs } from '@datadog/browser-logs'

import PaginationWidget from './PaginationWidget.vue'

jest.mock('@datadog/browser-logs')

function renderComponent(props = {}) {
  return mount(PaginationWidget, {
    props,
  })
}

describe('PaginationWidget.vue', () => {
  it('renders nothing if no next or prev props provided', () => {
    const wrapper = renderComponent()

    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders pagination', () => {
    const wrapper = renderComponent({
      hasPrevious: true,
      hasNext: true,
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('calls buttons', async () => {
    const wrapper = renderComponent({
      hasPrevious: true,
      hasNext: true,
    })

    await wrapper.find('[data-testid="pagination-previous-button"]').trigger('click')
    await wrapper.find('[data-testid="pagination-next-button"]').trigger('click')
    expect(datadogLogs.logger.info).toMatchSnapshot()
  })
})
