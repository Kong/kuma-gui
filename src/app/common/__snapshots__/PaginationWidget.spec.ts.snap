// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationWidget.vue calls buttons 1`] = `
[MockFunction] {
  "calls": [
    [
      "pagination-previous-button-clicked",
    ],
    [
      "pagination-next-button-clicked",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`PaginationWidget.vue renders nothing if no next or prev props provided 1`] = `
<div>
  <div
    class="pagination"
  >
    <!--v-if-->
    <!--v-if-->
  </div>
</div>
`;

exports[`PaginationWidget.vue renders pagination 1`] = `
<div>
  <div
    class="pagination"
  >
    <button
      class="medium rounded primary k-button"
      data-v-6f773baa=""
      type="button"
    >
      
      <!---->
      
      
       ‹ Previous 
      
      <!---->
    </button>
    <button
      class="medium rounded primary k-button"
      data-v-6f773baa=""
      type="button"
    >
      
      <!---->
      
      
       Next › 
      
      <!---->
    </button>
  </div>
</div>
`;
